src/Psl/Comparison/equal.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Comparison;
+
+/**
+ * @template T
+ *
+ * @param T $a
+ * @param T $b
+ */
+function equal(mixed $a, mixed $b): bool
+{
+    return compare($a, $b) === Order::Equal;
```

Comment:
I was thinking about adding a `Equable` check here and use the `Equable::equals()` method if possible. This could work for equal and not equal.
However, it would mean adding additional comparisons for `less_or_equal` and `greater_or_equal`. Not sure if it's a good idea and worth adding.